apps/namadillo/src/hooks/useExtensionEvents.tsx

```diff
@@ -16,6 +21,7 @@ export const useExtensionEvents = (): void => {
 
   // Register handlers:
   useEventListenerOnce(Events.AccountChanged, async () => {
+    await accounts.refetch();
```

Comment:
This fixes two cases:

- When the selected account in Extension is a spending key import (no transparent account), and
- When a Shielded account changes, we need to refetch accounts so when it's linked to the parent Transparent account, the details will be correct